### PR TITLE
Add Blood Glucose, Body Fat, Body Temperaure, and Basal Metabolic Rate sensors

### PR DESCRIPTION
**Pull request by lunarok**

### CODE REVIEW

File: custom_components/google_fit/api.py [hunk 1]
```diff
@@ -95,10 +95,13 @@ def __init__(self):
             lastUpdate=datetime.now(),
             activeMinutes=None,
             calories=None,
+            basalMetabolicRate=None,
             distance=None,
             heartMinutes=None,
             height=None,
             weight=None,
+            fat=None,
```
Comment (YorkshireIoT) [inline custom_components/google_fit/api.py:103]:
```suggestion
            bodyFat=None,
```

Where possible it's good to match what Google refers to the sensor as

File: custom_components/google_fit/api.py [hunk 2]
```diff
@@ -95,10 +95,13 @@ def __init__(self):
...
+            temperature=None,
```
Comment (YorkshireIoT) [inline custom_components/google_fit/api.py:104]:
```suggestion
            bodyTemperature=None,
```
Similarly for body temperature

File: custom_components/google_fit/const.py [hunk 3]
```diff
@@ -117,6 +130,26 @@
         source="derived:com.google.weight:com.google.android.gms:merge_weight",
         data_key="weight",
     ),
+    GoogleFitSensorDescription(
+        key="google_fit",
+        name="Fat",
```
Comment (YorkshireIoT) [inline custom_components/google_fit/const.py:135]:
```suggestion
        name="Body Fat",
```

File: custom_components/google_fit/const.py [hunk 4]
```diff
@@ -117,6 +130,26 @@
...
+        icon="mdi:scale-balance",
+        native_unit_of_measurement=PERCENTAGE,
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=None,
+        source="derived:com.google.body.fat.percentage:com.google.android.gms:merged",
+        data_key="fat",
+    ),
+    GoogleFitSensorDescription(
+        key="google_fit",
+        name="Temperature",
```
Comment (YorkshireIoT) [inline custom_components/google_fit/const.py:145]:
```suggestion
        name="Body Temperature",
```

**Comment by lunarok** [comment]
Yes, look better, actually didn't see that Github has forwarded the commits done after the PR, was expecting to push first only glucose and test the others first
But tests are done and it works on my side, but names are better in proposition yes

File: custom_components/google_fit/api.py [hunk 5]
```diff
@@ -266,9 +270,13 @@ def _parse_point(self, request_id: str, response: FitnessDataPoint) -> None:
         if request_id in [
             "height",
             "weight",
+            "basalMetabolicRate",
+            "fat",
```
Comment (YorkshireIoT) [inline custom_components/google_fit/api.py:274]:
```suggestion
            "bodyFat",
```

File: custom_components/google_fit/api.py [hunk 6]
```diff
@@ -266,9 +270,13 @@ def _parse_point(self, request_id: str, response: FitnessDataPoint) -> None:
...
+            "temperature",
```
Comment (YorkshireIoT) [inline custom_components/google_fit/api.py:275]:
```suggestion
            "bodyTemperature",
```

File: custom_components/google_fit/api_types.py [hunk 7]
```diff
@@ -21,10 +21,13 @@ class FitnessData(TypedDict):
     lastUpdate: datetime
     activeMinutes: float | None
     calories: float | None
+    basalMetabolicRate: float | None
     distance: float | None
     heartMinutes: float | None
     height: float | None
     weight: float | None
+    fat: float | None
```
Comment (YorkshireIoT) [inline custom_components/google_fit/api_types.py:29]:
```suggestion
    bodyFat: float | None
```

File: custom_components/google_fit/api_types.py [hunk 8]
```diff
@@ -21,10 +21,13 @@ class FitnessData(TypedDict):
...
+    temperature: float | None
```
Comment (YorkshireIoT) [inline custom_components/google_fit/api_types.py:30]:
```suggestion
    bodyTemperature: float | None
```

File: custom_components/google_fit/const.py [hunk 9]
```diff
@@ -117,6 +130,26 @@
...
     ),
+    GoogleFitSensorDescription(
+        key="google_fit",
+        name="Fat",
+        icon="mdi:scale-balance",
+        native_unit_of_measurement=PERCENTAGE,
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=None,
+        source="derived:com.google.body.fat.percentage:com.google.android.gms:merged",
+        data_key="fat",
```
Comment (YorkshireIoT) [inline custom_components/google_fit/const.py:141]:
```suggestion
        data_key="bodyFat",
```

File: custom_components/google_fit/const.py [hunk 10]
```diff
@@ -117,6 +130,26 @@
...
+    ),
+    GoogleFitSensorDescription(
+        key="google_fit",
+        name="Temperature",
+        icon="mdi:thermometer",
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        source="derived:com.google.body.temperature:com.google.android.gms:merged",
+        data_key="temperature",
```
Comment (YorkshireIoT) [inline custom_components/google_fit/const.py:151]:
```suggestion
        data_key="bodyTemperature",
```

File: custom_components/google_fit/const.py [hunk 11]
```diff
@@ -77,6 +80,16 @@
...
     ),
+    GoogleFitSensorDescription(
+        key="google_fit",
+        name="Basal Metabolic Rate",
+        icon="mdi:target",
+        native_unit_of_measurement="kcal",
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=None,
+        source="derived:com.google.calories.bmr:com.google.android.gms:merged",
+        data_key="basalMetabolicRate",
```
Comment (YorkshireIoT) [inline custom_components/google_fit/const.py:91]: You should also add this and the other new sensors to the sensor table in the README at the root of the repo

**Review by YorkshireIoT — CHANGES_REQUESTED** [review]

File: custom_components/google_fit/const.py (hunk 11 above)
Comment (lunarok) [reply custom_components/google_fit/const.py:91]: Done : https://github.com/YorkshireIoT/ha-google-fit/pull/74/commits/6ca1a47f4b66710f4386dc71d0e5ec83109bbd06

File: README.md [hunk 12]
```diff
@@ -24,6 +24,8 @@ Platform | Name |Description
 `sensor` | `heart_points_daily` | [Heart Points][heart-points] earned. Reset daily.
 `sensor` | `height` | [Height][height] (metres).
 `sensor` | `weight` | [Weight][weight] (kilograms).
+`sensor` | `fat` | [Body Fat][fat] (percentage).
```
Comment (YorkshireIoT) [inline README.md:27]:
```suggestion
`sensor` | `body_fat` | [Body Fat][fat] (percentage).
```
The sensor name should match the automatic name Home Assistant gives it which is lowercase, snakecase

File: README.md [hunk 13]
```diff
@@ -24,6 +24,8 @@ Platform | Name |Description
 `sensor` | `heart_points_daily` | [Heart Points][heart-points] earned. Reset daily.
 `sensor` | `height` | [Height][height] (metres).
 `sensor` | `weight` | [Weight][weight] (kilograms).
+`sensor` | `fat` | [Body Fat][fat] (percentage).
+`sensor` | `temperature` | [Body Temperature][temperature] (celcius).
```
Comment (YorkshireIoT) [inline README.md:28]:
```suggestion
`sensor` | `body_temperature` | [Body Temperature][temperature] (celcius).
```

File: README.md (hunk 12 above)
Comment (YorkshireIoT) [inline README.md:27]: You need to add relevant links for the new sensors so [fat] actually links to something. See what's been done for the others at the bottom of this file

File: custom_components/google_fit/const.py (hunk 11 above)
Comment (YorkshireIoT) [reply custom_components/google_fit/const.py:91]: You still haven't added basal metabolic rate to the README

**Review by YorkshireIoT — CHANGES_REQUESTED** [review]

File: README.md (hunk 12 above)
Comment (lunarok) [reply README.md]: Done https://github.com/YorkshireIoT/ha-google-fit/pull/74/commits/1318b9b7f971cf9b02c1b730210bd9c61309bff8

File: custom_components/google_fit/const.py (hunk 11 above)
Comment (lunarok) [reply custom_components/google_fit/const.py:91]: Done https://github.com/YorkshireIoT/ha-google-fit/pull/74/commits/1318b9b7f971cf9b02c1b730210bd9c61309bff8

**Comment by YorkshireIoT** [comment]
Thank you for all your work @lunarok 😄

**Pull request merged**